app/services/nsm/all_adjustments_deleter.rb

```diff
@@ -10,12 +10,53 @@ def initialize(params, adjustment_type, current_user)
     def call
       return false unless submission.editable_by?(current_user)
 
-      raise StandardError, "no adjustments to delete for id:#{submission.id}" unless submission.any_adjustments?
+      delete_work_item_adjustments if work_items
+      delete_letters_and_calls_adjustments if letters_and_calls
+      delete_disbursement_adjustments if disbursements
+      ::Event::DeleteAdjustments.build(submission:, comment:, current_user:)
+      submission.save!
+    end
+
+    def delete_work_item_adjustments
+      revert_fields = %w[uplift pricing work_type time_spent]
+      work_items.each do |work_item|
+        revert_fields.each do |field|
+          revert(work_item, field, 'work_items')
+        end
+        work_item.delete('adjustment_comment')
+      end
+    end
 
-      app_store_record = AppStoreClient.new.get_submission(submission)
-      submission.update!(data: app_store_record['application'])
+    def delete_letters_and_calls_adjustments
+      revert_fields = %w[uplift count]
+      letters_and_calls.each do |letter_or_call|
+        revert_fields.each do |field|
+          revert(letter_or_call, field, letter_or_call['type']['value'])
+          letter_or_call.delete('adjustment_comment')
```

Comment:
Why is this one inside the `revert_fields.each` but the other 2 are outside?

Reply:
Good spot! I don't think it should be here, so have moved.